src/views/QuestionnaireEdit/SelectOneQuestionForm/index.tsx

```diff
@@ -0,0 +1,278 @@
+                if (!response) {
+                    return;
+                }
+                if (response.ok) {
+                    alert.show(
+                        'Question created successfully.',
+                        { variant: 'success' },
+                    );
+                } else {
+                    alert.show(
+                        'Failed to create question.',
+                        { variant: 'error' },
+                    );
+                }
+            },
+            onError: () => {
+                alert.show(
+                    'Failed to create question.',
+                    { variant: 'error' },
+                );
+            },
+        },
+    );
+    const initialFormValue: FormType = {
+        type: 'SELECT_ONE' as QuestionTypeEnum,
+        questionnaire: questionnaireId,
+        name: randomString(),
+    };
+
+    const {
+        pristine,
+        validate,
+        value: formValue,
+        error: formError,
+        setFieldValue,
+        setError,
+    } = useForm(schema, { value: initialFormValue });
+
+    const fieldError = getErrorObject(formError);
+
+    const handleQuestionSubmit = useCallback(() => {
+        const handler = createSubmitHandler(
+            validate,
+            setError,
+            (valueFromForm) => {
+                triggerQuestionCreate({
+                    variables: {
+                        projectId,
+                        input: valueFromForm as QuestionCreateInput,
+                    },
+                });
+            },
+        );
+        handler();
+    }, [
+        triggerQuestionCreate,
+        projectId,
+        setError,
+        validate,
+    ]);
+
+    return (
+        <form className={styles.question}>
+            <SelectOneQuestionPreview
+                className={styles.preview}
+                label={formValue.label}
+                hint={formValue.hint}
+            />
+            <div className={styles.editSection}>
+                <TextInput
+                    name="label"
+                    label="Question label"
+                    value={formValue.label}
+                    error={fieldError?.label}
+                    onChange={setFieldValue}
+                />
+                <TextInput
+                    name="hint"
+                    label="Hint"
+                    value={formValue.hint}
+                    error={fieldError?.hint}
+                    onChange={setFieldValue}
+                />
+                <>Options</>
+                <SelectInput
+                    name="label"
+                    label="Pillar and Sub pillar"
+                    value={formValue.label}
+                    error={fieldError?.label}
+                    onChange={setFieldValue}
+                    keySelector={pillarKeySelector}
+                    labelSelector={pillarLabelSelector}
+                    options={pillarsOptions}
+                />
```

Comment:
I think we should make a separate component for this SelectInput. Also, let's make this a SearchSelectInput.